### PR TITLE
[Train][Workspace template] Fix some LLM finetuning WS issues.

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?

**1. Updated the Readme:**
* Added time to run fields for all models (after running all variations at least once)
* Fix CLI typo
* Added guideline on how to pick the instance types
 
**2. Updated the cluster config**
* Base image to 2.6.1 and cuda117
* use pip package list instead of post build commands
* added debian packages for enabling nvme offloading

**3. Added and modified compute configs**
* Added two versions of compute config for 70B
* Modified the other compute configs to include custom resourse

**4. Updated Deepspeed configs**

**5. Modified the fine-tuning script**
* padding=`longest` --> padding=`max_length`: This will ensure we are measuring the perf. consistently with a fixed block size
* Evaluation method had some bugs around aggregation that are fixed 
* Added `ACCELERATE_TORCH_DEVICE` setup directly into the training function until this is fixed by ray train
* Moved the tmp directory for checkpoint from root to output dir which allows it to be on nvme: This opens up capacity for more disk when checkpointing a model as big as 70B.
* Enabled sharding for evaluation dataset which speeds up evaluation by 16x for 7/13B and 32x for 70B.

**6. Fix the `--as-test` flag bugs in the shell scripts**
**7. Consolidated all shell scripts into one to remove code-duplication**

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/templates/04_finetuning_llms_with_deepspeed/compute_configs/aws_70b_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+
+
+
+cloud_id: {{env["ANYSCALE_CLOUD_ID"]}}
+region: us-west-2
```
Comment (justinvyu) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/compute_configs/aws_70b_v1.yaml:6]: These cannot have cloud id and region.

Comment (justinvyu) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/compute_configs/aws_70b_v1.yaml:6]: These cannot have cloud id and region (this is not a valid yaml file and will error when product tries to parse it).

File: doc/source/templates/04_finetuning_llms_with_deepspeed/cluster_env.yaml [hunk 2]
```diff
@@ -1,31 +1,33 @@
 # See https://hub.docker.com/r/anyscale/ray for full list of
 # available Ray, Python, and CUDA versions.
-base_image: anyscale/ray:2.5.1-py39-cu116
+base_image: {{ env["RAY_IMAGE_ML_NIGHTLY_GPU"] | default("anyscale/ray:2.6.1-py39-cu117") }}
```
Comment (justinvyu) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/cluster_env.yaml:3]: This is also not usable by the customer -- should this be a private yaml file under `release/` rather than in the template working dir?

**Review by justinvyu — CHANGES_REQUESTED** [review]

File: doc/source/templates/04_finetuning_llms_with_deepspeed/cluster_env.yaml (hunk 2 above)
Comment (kouroshHakha) [reply doc/source/templates/04_finetuning_llms_with_deepspeed/cluster_env.yaml:3]: yeah these are the stuff that I was trying to get the releas to work. I'll revert these.

**Review by justinvyu — APPROVED** [review]
